src/main/resources/frontend/src/services/adminService.ts

```diff
@@ -1,112 +1,44 @@
-import {useAppConfigStore} from "@/stores/appConfig";
 import type {CanvasSection, Phase, Submission, User } from '@/types/types'
 import type {Option} from "@/views/AdminView/Analytics.vue";
+import { ServerCommunicator } from '@/network/ServerCommunicator'
 
 export const usersGet = async (): Promise<User[]> => {
     try {
-        const response = await fetch(useAppConfigStore().backendUrl + '/api/admin/users', {
-            method: 'GET',
-            credentials: 'include'
-        });
-
-        return await response.json();
+        return await ServerCommunicator.getRequest<User[]>('/api/admin/users')
     } catch (e) {
         return [];
```

Comment:
I'm noticing that this `try ... catch ... return default value` is a common pattern among all these endpoints. A few of them go a step further by also logging an console message before returning the default value.

Can we create an overload or similar method that streamlines this process by also taking in a default value to respond to an error in the request? Potentially, the argument could look like: `onError: T | ((error: Error) => T)`. This would allow a user of the function to drop in a default value to return, **or** provide a function that would perform custom logic before returning a default value.

Reply:
As I mentioned in another comment, I did play around with it. I'll go ahead and add back in the `defaultReturn` optional parameter

Reply:
Any update on this?